### ISSUE TITLE
When running mir-test-tools.performance-test on Xorg (Mir-on-X) only one GLMark2Wayland.* test passes

### ISSUE THREAD

**Issue by AlanGriffiths** [MirServer/mir#1921]
Run:
```plain
mir-test-tools.performance-test --gtest_filter=-GLMark2Xwayland*
```

Expect: all tests pass
Actual: the second test (GLMark2Xwayland.windowed) fails or segfaults.

Further investigation shows that only one of `GLMark2Wayland.fullscreen` or `GLMark2Wayland.windowed` will pass and only once (i.e. both fail the second time with `--gtest_repeat`, even though all the other tests pass between.)

Running the same tests from the .deb install also fails, but far less frequently.

The same happens with Xvfb:
```plain
sudo apt install xvfb
Xvfb :99&
DISPLAY=:99 mir-test-tools.performance-test --gtest_filter=-GLMark2Xwayland*
```

**Comment by AlanGriffiths** [comment]
# Updated test scenario

For when someone gets to it:
```
sudo apt install xvfb
Xvfb :99&
DISPLAY=:99 mir_performance_tests --gtest_filter=GLMark2Wayland.* --gtest_repeat=100
```

**Comment by AlanGriffiths** [comment]
May be related to #1704

**Comment by AlanGriffiths** [comment]
This is no longer a valid scenario due to changes to the X11 snap interface: We can no longer run these tests on Mir on X as connecting to the host conflicts with hosting our own X11 session.

The nearest scenario would be hosting on Wayland...

* `snap disconnect mir-test-tools:x11` - disconnect the X11 interface (enabling serving X11)
* `mir-test-tools.performance-test --gtest_filter=GLMark2Wayland.* --gtest_repeat=100 --wayland-host $WAYLAND_DISPLAY` - run using Mir-on-Wayland

That seems to work, and problems relating to the x11 platform are addressed by #1704, so closing.
